libmamba/include/mamba/core/problems_graph_creator.hpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef MAMBA_PROBLEMS_GRAPH_CREATOR_HPP
+#define MAMBA_PROBLEMS_GRAPH_CREATOR_HPP
+
+#include <string>
+#include <utility>
+#include <vector>
+#include <regex>
+#include <variant>
+#include <unordered_set>
```

Comment:
It looks like we don't need all these headers in the `hpp` files
```suggestion
#include <string>
#include <utility>
#include <vector>
#include <variant>
#include <unordered_set>
```